### ISSUE TITLE
RuntimeError after optimize_for_mobile

### ISSUE THREAD

**Issue by myuanqing** [pytorch/pytorch#57271]
Hi

I am trying to use **optimize_for_mobile** from **torch.utils.mobile_optimizer** to optimize LSTM. However, after being optimized, the model won't work. The error message is 
![image](https://user-images.githubusercontent.com/40328758/116532425-cf752b00-a912-11eb-8481-ebb9cb6e2093.png)
I am using pytorch1.6.0+cu101, the model to be optimized is from an older version. I am wondering if it is the problem of compatibility.

Could anyone help me with this issue? Thanks in advance.

**Comment by guangy10** [comment]
Thanks for reporting the issue. Can you provide a bit more information about your setup? We will be looking into it. cc: @kimishpatel @JacobSzwejbka

**Comment by kimishpatel** [comment]
If you can provide minimal repro that would be great.

**Comment by myuanqing** [comment]
> If you can provide minimal repro that would be great.

Shall I attach the model file? Seems that a file bigger than 10 MB cannot be posted, so I upload the file to which has already been applied "optimized_for_mobile". 
[model file.zip](https://github.com/pytorch/pytorch/files/6403543/model.file.zip)

I have found another weird thing: when applied optimized_for_mobile in my PC's windows environment with pytorch1.6+cpu, the size of LSTM model isn't reduced, while applied optimized_for_mobile in the server's linux environment with pytorch1.6+cu101, model file goes from 50MB to 1.6MB. （PS: Would a non-optimized LSTM model be larger than 50 MB? ）

**Comment by kimishpatel** [comment]
50MB > 1.6MB is not really expected. Is 50MB model also quantized?

Did you also try to run this optimized model on windows? I wonder if there is a different behavior.

For the model, I will take a look.

**Comment by myuanqing** [comment]
> 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> 
> Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> 
> For the model, I will take a look.

The 50MB model is not quantized.

I tried the 1.6MB on windows, it didn't work neither.

**Comment by kimishpatel** [comment]
> > 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> > Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> > For the model, I will take a look.
> 
> The 50MB model is not quantized.
> 
> I tried the 1.6MB on windows, it didn't work neither.

Can you provide the quantized model that was before optimize_for_mobile and after?

**Comment by myuanqing** [comment]
> > > 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> > > Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> > > For the model, I will take a look.
> > 
> > 
> > The 50MB model is not quantized.
> > I tried the 1.6MB on windows, it didn't work neither.
> 
> Can you provide the quantized model that was before optimize_for_mobile and after?

Of course. But it seems that a file larger than 10 MB  is supported to be uploaded on github. Could I send it later to you by email?

**Comment by kimishpatel** [comment]
> > > > 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> > > > Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> > > > For the model, I will take a look.
> > > 
> > > 
> > > The 50MB model is not quantized.
> > > I tried the 1.6MB on windows, it didn't work neither.
> > 
> > 
> > Can you provide the quantized model that was before optimize_for_mobile and after?
> 
> Of course. But it seems that a file larger than 10 MB is supported to be uploaded on github. Could I send it later to you by email?

How big is it? Can you share with dropbox folder or something similar?

**Comment by myuanqing** [comment]
> > > > > 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> > > > > Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> > > > > For the model, I will take a look.
> > > > 
> > > > 
> > > > The 50MB model is not quantized.
> > > > I tried the 1.6MB on windows, it didn't work neither.
> > > 
> > > 
> > > Can you provide the quantized model that was before optimize_for_mobile and after?
> > 
> > 
> > Of course. But it seems that a file larger than 10 MB is supported to be uploaded on github. Could I send it later to you by email?
> 
> How big is it? Can you share with dropbox folder or something similar?

Sorry for my late reply，I just took a holiday. 

Do you mean the .pt file? It is 50MB. I put it onto [https://www.dropbox.com/s/fashvm4pk7rt5mq/olddec.pt?dl=0]( https://www.dropbox.com/s/fashvm4pk7rt5mq/olddec.pt?dl=0 ), could you get access to the file?

**Comment by kimishpatel** [comment]
> > > > > > 50MB > 1.6MB is not really expected. Is 50MB model also quantized?
> > > > > > Did you also try to run this optimized model on windows? I wonder if there is a different behavior.
> > > > > > For the model, I will take a look.
> > > > > 
> > > > > 
> > > > > The 50MB model is not quantized.
> > > > > I tried the 1.6MB on windows, it didn't work neither.
> > > > 
> > > > 
> > > > Can you provide the quantized model that was before optimize_for_mobile and after?
> > > 
> > > 
> > > Of course. But it seems that a file larger than 10 MB is supported to be uploaded on github. Could I send it later to you by email?
> > 
> > 
> > How big is it? Can you share with dropbox folder or something similar?
> 
> Sorry for my late reply，I just took a holiday.
> 
> Do you mean the .pt file? It is 50MB. I put it onto https://www.dropbox.com/s/fashvm4pk7rt5mq/olddec.pt?dl=0, could you get access to the file?

Sorry, since last time we spoke I havent had time to look into this yet.
